packages/bsky/src/hydration/util.ts

```diff
@@ -164,3 +164,23 @@ export const isActivitySubscriptionEnabled = ({
   post: boolean
   reply: boolean
 }): boolean => post || reply
+
+export function isDebugFieldAllowed(
+  allowedDids: Set<string>,
+  did?: string | null,
+) {
+  if (!did) return false
+  return allowedDids.has(did)
+}
+
+export type DebugFieldObject = { [key: string]: unknown } | undefined
+
+export function mergeDebugFieldObjects(debugs: DebugFieldObject[]) {
+  const merged: DebugFieldObject = {}
+  for (const debug of debugs) {
+    if (!!debug && typeof debug === 'object') {
```

Comment:
I think type narrowing has our back here.
```suggestion
    if (debug) {
```